servers/src/grin/sync/header_sync.rs

```diff
@@ -66,9 +66,6 @@ impl HeaderSync {
 					header_head.height,
 				);
 
-				// Reset sync_head to the same as current header_head.
-				self.chain.reset_sync_head(&header_head).unwrap();
```

Comment:
Why we remove the `reset_sync_head` here?  it broke those syncing logic we have discussed heavily before.

Reply:
Hmm. This may have been an unintentional change when we were cleaning up the "reset head" code.
We _do_ need something here don't we.